Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -54,17 +54,20 @@ class ChunkedFileUploader {
             currentWorkTask = nil
             notifyStateFromMain(.paused(update))
         }
-        default: do {}
+        case .ready, .paused, .canceled, .success, .failure:
+            break
         }
     }
     
     /// Starts the upload if it wasn't already starting
     func start() {
         switch _currentState {
-        case .ready: fallthrough
-        case .paused(_):
+        case .ready:
+            _currentState = .starting
             beginUpload()
-        default:
+        case .paused:
+            beginUpload()
```

Comment:
When resuming an upload, setting the `_currentState` to starting may cause inaccurate progress reporting if the upload is paused a second time in the time between the transition between `.starting` and `.uploading`.